### PR TITLE
GEOMESA-1677 Spark SQL: deal with null arguments to UDFs

### PR DESCRIPTION
**Pull request by mdzimmerman**
Adds a wrapper around most UDFs that has each function return
null if one or more arguments are null, rather than throwing
NPEs or other errors that prevents whole queries from executing.

Also adds tests for cast functions, which were omitted in test suite,
and removes loading of data in test classes where those data were not
used.

Signed-off-by: Matthew Zimmerman <matt.zimmerman@ccri.com>

### CODE REVIEW

File: geomesa-spark/geomesa-spark-sql/src/main/scala/org/apache/spark/sql/SQLFunctionHelper.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package org.apache.spark.sql
+
+/**
```
Comment (aheyne) [inline geomesa-spark/geomesa-spark-sql/src/main/scala/org/apache/spark/sql/SQLFunctionHelper.scala]: License header and created by block

File: geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometricCastTest.scala [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import org.apache.spark.sql._
+import org.geotools.data.{DataStoreFinder, DataStore}
+import org.junit.runner.RunWith
+import org.locationtech.geomesa.utils.interop.WKTUtils
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+import scala.collection.JavaConversions._
+
+/**
```
Comment (aheyne) [inline geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometricCastTest.scala]: Same, header and comment block

File: geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometryProcessingFunctionsTest.scala [hunk 3]
```diff
@@ -43,31 +43,45 @@ class SparkSQLGeometryProcessingFunctionsTest extends Specification with LazyLog
...
+          """.stripMargin)
+        buf.
```
Comment (aheyne) [inline geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometryProcessingFunctionsTest.scala]: strange new line

File: geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometryProcessingFunctionsTest.scala [hunk 4]
```diff
@@ -79,14 +93,21 @@ class SparkSQLGeometryProcessingFunctionsTest extends Specification with LazyLog
           |)
           |where
           |   size(l) > 1
-        """.stripMargin).collect().map(_.getDouble(2))
-      Array(70681.00230533161,141178.0595870766) must beEqualTo(res)
+        """.stripMargin).
+          collect().map(_.getDouble(2))
+      Array(
```
Comment (aheyne) [inline geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometryProcessingFunctionsTest.scala]: Another odd new line

File: geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometryProcessingFunctionsTest.scala [hunk 5]
```diff
@@ -98,12 +119,13 @@ class SparkSQLGeometryProcessingFunctionsTest extends Specification with LazyLog
           |)
           |where
           |   size(l) > 1
-        """.stripMargin).collect().map(_.getDouble(1))
-      Array(70681.00230533161,141178.0595870766) must beEqualTo(res)
+        """.stripMargin).
+          collect().map(_.getDouble(1))
+      Array(
+        70681.00230533161, 141178.0595870766) must beEqualTo(res)
```
Comment (aheyne) [inline geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometryProcessingFunctionsTest.scala]: same

File: geomesa-spark/geomesa-spark-sql/src/main/scala/org/apache/spark/sql/SQLFunctionHelper.scala (hunk 1 above)
Comment (mdzimmerman) [inline geomesa-spark/geomesa-spark-sql/src/main/scala/org/apache/spark/sql/SQLFunctionHelper.scala]: Oh license headers, my old nemesis!

File: geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometricCastTest.scala (hunk 2 above)
Comment (mdzimmerman) [inline geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometricCastTest.scala]: Fixed.

File: geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometryProcessingFunctionsTest.scala (hunk 3 above)
Comment (mdzimmerman) [inline geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometryProcessingFunctionsTest.scala]: Looks like IntelliJ made some weird decisions when it reformatted the code! Fixed.

File: geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometryProcessingFunctionsTest.scala (hunk 4 above)
Comment (mdzimmerman) [inline geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometryProcessingFunctionsTest.scala]: Fixed.

File: geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometryProcessingFunctionsTest.scala (hunk 5 above)
Comment (mdzimmerman) [inline geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometryProcessingFunctionsTest.scala]: Fixed.

File: geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometryProcessingFunctionsTest.scala [hunk 6]
```diff
@@ -43,31 +43,44 @@ class SparkSQLGeometryProcessingFunctionsTest extends Specification with LazyLog
...
     "st_antimeridianSafeGeom" >> {
-      val geom = "st_geomFromWKT('POLYGON((-190 50, -190 60, -170 60, -170 50, -190 50))')";
-      val decomposed = sc.sql(s"select st_antimeridianSafeGeom($geom)").collect.head.get(0).asInstanceOf[Geometry]
-      decomposed mustEqual WKTUtils.read(
-        "MULTIPOLYGON (((-180 50, -180 60, -170 60, -170 50, -180 50)), ((180 60, 180 50, 170 50, 170 60, 180 60)))")
+      "should handle nulls" >> {
+        sc.sql("select st_antimeridianSafeGeom(null)").collect.head(0) must beNull
+      }
+
+      "should split a geom that spans the antimeridian" >> {
```
Comment (jnh5y) [inline geomesa-spark/geomesa-spark-sql/src/test/scala/org/locationtech/geomesa/spark/SparkSQLGeometryProcessingFunctionsTest.scala]: Should we verify that the st_antimeridianSafeGeom code is a no-op away from the anti-meridian?  (Separate test?)

**Pull request merged**